src/app/search-users/search-users.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { SkillsService } from '../skills/skills.service';
+import { Skill } from '../skills/skill';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Observable, of } from 'rxjs';
+import { SearchUsersService } from './search-users.service';
+import { AnonymousUserSkill } from './anonymous-user-skill';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+
+@Component({
+  selector: 'app-search-users',
+  templateUrl: './search-users.component.html',
+  styleUrls: ['./search-users.component.scss']
+})
+export class SearchUsersComponent implements OnInit {
+
+  skills$: Observable<Skill[]> = of([]);
+  userSkills: AnonymousUserSkill[] = [];
+  showSearchResult = false;
+  errorMessage: string = null;
+
+  public form: FormGroup;
+
+  constructor(private skillsService: SkillsService,
+              private searchService: SearchUsersService,
+              private fb: FormBuilder,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService) {
+  }
+
+  ngOnInit() {
+    this.skills$ = this.skillsService.getAllSkills();
+    this.buildForm();
+  }
+
+  createCriteria(): FormGroup {
+    return this.fb.group({
+      skill: [null, Validators.required],
+      level: 0 // default skill level
+    });
+  }
+
+  addCriteria() {
+    this.criteriaList.push(this.createCriteria());
+  }
+
+  removeCriteria(index) {
+    this.criteriaList.removeAt(index);
+    this.checkDuplicates();
+  }
+
+  checkDuplicates() {
+    const counts = [];
+
+    this.criteriaList.controls.forEach(item => {
+      if (item.value.skill === null || counts[item.value.skill] === undefined) {
```

Comment:
`item.value.skill === null` can be replaced with `!item.value.skill` to make check be more general.
`counts[item.value.skill] === undefined` can be replaced with `!counts[item.value.skill]` to make check be more general.